x-pack/legacy/plugins/ml/server/routes/filters.ts

```diff
@@ -0,0 +1,227 @@
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Filters
+   *
+   * @api {put} /api/ml/filters Creates a filter
+   * @apiName CreateFilter
+   * @apiDescription Instantiates a filter, for use by custom rules in anomaly detection.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filter created filter
+   */
+  router.put(
+    {
+      path: '/api/ml/filters',
+      validate: {
+        body: schema.object(createFilterSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const body = request.body;
+        const resp = await newFilter(context, body);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Filters
+   *
+   * @api {put} /api/ml/filters/:filterId Updates a filter
+   * @apiName UpdateFilter
+   * @apiDescription Updates the  description of a filter, adds items or removes items.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filter updated filter
+   */
+  router.put(
+    {
+      path: '/api/ml/filters/{filterId}',
+      validate: {
+        params: schema.object({ filterId: schema.string() }),
+        body: schema.object(updateFilterSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { filterId } = request.params;
+        const body = request.body;
+        const resp = await updateFilter(context, filterId, body);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Filters
+   *
+   * @api {delete} /api/ml/filters/:filterId Delete filter
+   * @apiName DeleteFilter
+   * @apiDescription Deletes the filter with the specified ID.
+   *
+   * @apiParam {String} filterId the ID of the filter to delete
+   */
+  router.delete(
+    {
+      path: '/api/ml/filters/{filterId}',
+      validate: {
+        params: schema.object({ filterId: schema.string() }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const { filterId } = request.params;
+        const resp = await deleteFilter(context, filterId);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Filters
+   *
+   * @api {get} /api/ml/filters/_stats Gets filters stats
+   * @apiName GetFiltersStats
+   * @apiDescription Retrieves the list of filters which are used for custom rules in anomaly detection,
+   *          with stats on the list of jobs and detectors which are using each filter.
+   *
+   * @apiSuccess {Boolean} success
```

Comment:
probably a copy-paste issue, `getAllFilterStats` returns an array

Reply:
Done in https://github.com/elastic/kibana/pull/57597/commits/0e08a2349214889364434dce9a2969bcbd56680d